### ISSUE TITLE
Lag, Blocking & visual glitches on firefox 

### ISSUE THREAD

**Issue by thet0ast3r** [polimediaupv/paella-core#347]
Hi, 

We recently switched to paella player 7 with opencast, and identified a range of problems with it, most likely (at least somewhat) related to each other. 

Issues that occur, most prominent on firefox:

* when quickly navigating with the playback bar, the playback bar lags behind in loading the video, cumulatively.

To reproduce: https://paellaplayer.upv.es/#/demos/hls-multiaudio-smil-1 play this video (using firefox), and quickly jump from the beginning to the end and back to the beginning with the mouse. After some quick repetitions, the player is behind and the playback bar starts jumping on its own. 

* on some systems (also mainly ff), when one tries to drag the playback bar, it sometimes gets stuck, resuming playback from an unwanted position (very undesirable in long videos) 
* sometimes the 'mouseup' seems to not be registered when using the playback bar. This results in the user unwantedly changing the playback position by moving the mouse after intended seeking. 
* hovering over timeline previews quickly can cause a large number of aborted requests for preview images (often multiple of the same request url) 
![example of lots of aborted requests ](https://github.com/polimediaupv/paella-core/assets/17761958/0a3dd755-20b0-46cf-b4c2-1480173c5fda)
* preview images tend to get stuck when moving the mouse outside of the player container quickly/to the side. They then block the view of the video feed(s). 

On my system, these issues only have a minor impact on usability (firefox 120, mac os 14.1.0, apple m1 pro), but we got reports from users that said it was nearly impossible to use the player (firefox on linux). 
I sadly was unable to pinpoint the exact cause of the problem, but there definitely seems to be some type of unintended behavior that limits the seekability of videos. 

Additional info: 
* Mostly videos with 2 streams are affected.
* Our videos are served via hls(wowza) and we use opencast-paella-player.
* single-stream videos without timeline previews seem to work mostly fine (also hls with wowza). 
* users that had problems with our videos also had problems with the [paella demo video](https://paellaplayer.upv.es/#/) abeit to a lesser extent (maybe/probably due to the short video and lesser amount of timeline previews)

Please let me know if you need any additional/specific information.

thanks :)

**Comment by karendolan** [comment]
@thet0ast3r in case it helps, our site uses these additional HLS.js params to help with buffering.

  lowLatencyMode: true,
  backBufferLength: 90,
  startFragPrefetch: true,
  
Added to the `defaultHlsConfig`  list.  https://github.com/polimediaupv/paella-core/blob/main/src/js/videoFormats/es.upv.paella.hlsVideoFormat.js#L11C14-L57
(FYI - Some of the params on the current defaultHlsConfig list are not longer used in the latest HLS.js)

**Comment by thet0ast3r** [comment]
ok, i did some testing with loglevel debug:

it seems to be related to the way the player syncs. Only after the last 'video synchronization triggered' appears, the playback bar returns to normal. 

This is what it looks like, is it unusual? 
![image](https://github.com/polimediaupv/paella-core/assets/17761958/7f2554c9-6672-4dcf-918c-ecceee945aad)

so if i change the readystate to something higher than 2, i might alleviate these ui issues?
To me it looks like the playback bar is coupled to the actual current position in the video, and it also appears to be that all events (moving/Dragging the bar) are being handled sequentially. Is there a way to just throw away (kind of debounce) all those unnecessary seeks?

Also: i thought i disabled the cookiedataplugin, but apparently it still writes the cookies very very often, i'll have to take a look at this, too.

**Comment by karendolan** [comment]
> so if i change the readystate to something higher than 2, i might alleviate these ui issues?
For our site, that worked. The sync code calls `setCurrentTime`
```
		this.player.log.debug("Video synchronization triggered");
		secPlayer.setCurrentTime(currentTime);
```
The `setCurrentTime(currentTime)` does a `waitForLoaded()`
```
        if (this._videoEnabled) {
            await this.waitForLoaded();
            return this.video.currentTime = t;
        }
```
The `waitForLoaded()` does the check `(this.video.readyState >= 2)`, as mentioned in my earlier post.

Giving the video extra time to buffer by reaching a higher readyState on every seek, preserves the seek sequence, and prevented undesired seek behavior for our site.

Ref for setCurrentTime's calls to waitForLoad() https://github.com/polimediaupv/paella-core/blob/main/src/js/videoFormats/es.upv.paella.mp4VideoFormat.js#L97-L106

**Comment by ferserc1** [comment]
La configuración de ejemplo del plugin de HLS estaba extraída de una versión bastante antigua de la biblioteca hls.js, que por lo visto en versiones recientes daba problemas.

Although it is not a paella-core issue in itself, I have modified some of these example configuration settings so as not to be misleading. However, I'm going to use the information in this thread to update the documentation about the HLS plugin

**Comment by ferserc1** [comment]
I have added a new parameter in the configuration to control the maximum desynchronization time of the videos. It is available starting from version 1.46.2
